frontend/src/pages/AllRecurringRunsList.test.tsx

```diff
@@ -0,0 +1,144 @@
+      updateToolbar: updateToolbarSpy,
+    };
+    return Object.assign(props, {
+      toolbarProps: new AllRecurringRunsList(props).getInitialToolbarState(),
+    });
+  }
+
+  function shallowMountComponent(
+    propsPatch: Partial<PageProps & { namespace?: string }> = {},
+  ): void {
+    tree = shallow(<AllRecurringRunsList {...generateProps()} {...propsPatch} />);
+    // Necessary since the component calls updateToolbar with the toolbar props,
+    // then expects to get them back in props
+    tree.setProps({ toolbarProps: _toolbarProps });
+    updateToolbarSpy.mockClear();
+  }
+
+  beforeEach(() => {
+    updateBannerSpy.mockClear();
+    updateToolbarSpy.mockClear();
+    historyPushSpy.mockClear();
+  });
+
+  afterEach(() => tree.unmount());
+
+  it('renders all recurring runs', () => {
+    shallowMountComponent();
+    expect(tree).toMatchInlineSnapshot(`
+      <div
+        className="page"
+      >
+        <RecurringRunList
+          history={
+            Object {
+              "push": [MockFunction],
+            }
+          }
+          location=""
+          match=""
+          onError={[Function]}
+          onSelectionChange={[Function]}
+          refreshCount={0}
+          selectedIds={Array []}
+          toolbarProps={
+            Object {
+              "actions": Object {
+                "newRun": Object {
+                  "action": [Function],
+                  "icon": [Function],
+                  "id": "createNewRunBtn",
+                  "outlined": true,
+                  "primary": true,
+                  "style": Object {
+                    "minWidth": 130,
+                  },
+                  "title": "Create run",
+                  "tooltip": "Create a new run",
+                },
+                "refresh": Object {
+                  "action": [Function],
+                  "id": "refreshBtn",
+                  "title": "Refresh",
+                  "tooltip": "Refresh the list",
+                },
+              },
+              "breadcrumbs": Array [],
+              "pageTitle": "Recurring Runs",
+            }
+          }
+          updateBanner={[MockFunction]}
+          updateDialog={[MockFunction]}
+          updateSnackbar={[MockFunction]}
+          updateToolbar={[MockFunction]}
+        />
+      </div>
+    `);
+  });
+
+  it('lists all recurring runs in namespace', () => {
+    shallowMountComponent({ namespace: 'test-ns' });
+    expect(tree.find('RecurringRunList').prop('namespaceMask')).toEqual('test-ns');
+  });
+
+  it('removes error banner on unmount', () => {
+    shallowMountComponent();
+    tree.unmount();
+    expect(updateBannerSpy).toHaveBeenCalledWith({});
+  });
+
+  // it('refreshes the recurring run list when refresh button is clicked', async () => {
+  //   const tree = render(<AllRecurringRunsList {...generateProps()} />);
+  //   await TestUtils.flushPromises()
+  //   fireEvent.click(tree.getByText('Refresh'));
+  // });
```

Comment:
Removing this commented block if test is not enabled, or add a TODO to explain why and plan for enabling it.

Reply:
@zijianjoy You are right. I added a TODO above the comment so we don't lose context and we can come back to it later on